regulations/views/partial_search.py

```diff
@@ -83,16 +85,25 @@ def get_context_data(self, **kwargs):
             result['header'] = node_types.label_to_text(result['label'])
             if 'title' in result:
                 result['header'] += ' ' + title(result['title'])
-            result['section_id'] = section_url.view_label_id(
-                result['label'], context['version'])
-            result['url'] = section_url.fetch(
-                result['label'], context['version'], sectional=True)
+            if doc_type == 'cfr':
+                result['section_id'] = section_url.view_label_id(
+                    result['label'], context['version'])
+                result['url'] = section_url.fetch(
+                    result['label'], context['version'], sectional=True)
+            else:
+                result['section_id'] = '-'.join(
+                    [result['label'][0], 'preamble'] + result['label'])
```

Comment:
I'm not sure if this is ever used, but won't it link too deeply into the hierarchy?